assemblies/assembly-planner-configuration.adoc

```diff
@@ -0,0 +1,1986 @@
+
+//:OPTAPLANNER-COMM:
+//Variable for the community version conditional statements.
+:OPTAPLANNER-ENT:
+//Variable for the enterprise version conditional statements.
+
+////
+ifdef::OPTAPLANNER-COMM[]
+[[plannerConfiguration]]
+= OptaPlanner configuration
+:doctype: book
+:imagesdir: ..
+:sectnums:
+:toc: left
+:icons: font
+:experimental:
+
+
+[[plannerConfigurationOverview]]
+== Overview
+endif::OPTAPLANNER-COMM[]
+////
+
+
+
+[id='assembly-planner-configuration_{context}']
+
+= The {PLANNER} solver
+
+Solving a planning problem with OptaPlanner consists of the following steps:
+
+. *Model your planning problem* as a class annotated with the `@PlanningSolution` annotation, for example the `NQueens` class.
+. *Configure a Solver*, for example a First Fit and Tabu Search solver for any `NQueens` instance.
+. *Load a problem data set* from your data layer, for example a Four Queens instance. That is the planning problem.
+. *Solve it* with `Solver.solve(problem)` which returns the best solution found.
+
+ifdef::OPTAPLANNER-COMM[]
+image::PlannerConfiguration/inputOutputOverview.png[align="center"]
+endif::OPTAPLANNER-COMM[]
+ifdef::OPTAPLANNER-ENT[]
+image::optimizer/inputOutputOverview.png[align="center"]
+endif::OPTAPLANNER-ENT[]
+
+
+
+
+
+== Configuring the {PLANNER} solver
+You can use the following methods to configure your OptaPlanner solver:
+
+* Use an XML file.
+* Use the `SolverConfig` API.
+* Add class annotations and JavaBean property annotations on the domain model.
+* Control the the method that OptaPlanner uses to access your domain.
+* Define custom properties.
+
+[[solverConfigurationByXML]]
+=== Using an XML file to configure the {PLANNER_SHORT} solver
+You can use an XML file to configure the solver. In a typical project that follows the Maven directory structure, the `solverConfig` XML file is located in the ``$PROJECT_DIR/src/main/resources/org/optaplanner/examples/_PROJECT_/solver`` directory, where `_PROJECT_` is the name of your OptaPlanner project. Alternatively, you can create a `SolverFactory` from a file with ``SolverFactory.createFromXmlFile()``.
```

Comment:
You need to change this section. So the context is that you have an empty project and you are trying to create a solver.
Starting from In a typical project there is no solverConfig yet, so you need to create it and provide within a classpath resource (as defined by ClassLoader.getResource()): 
$PROJECT_DIR/src/main/resources/.../nqueensSolverConfig.xml

Reply:
Or if you do mention here where to get the example configuration for project (with $PROJECT_DIR)better to mention it explicitly

Reply:
I've made some improvements, @dupliaka. Please have a look.

Reply:
LGTM